java/amazon-kinesis-producer-sample/src/com/amazonaws/services/kinesis/producer/sample/SampleProducer.java

```diff
@@ -111,6 +112,9 @@ public void onFailure(Throwable t) {
                                 last.getErrorCode(), last.getErrorMessage()));
                     }
 
+                } else if (t instanceof UnexpectedMessageException) {
```

Comment:
`int attempts = ((UserRecordFailedException) t).getResult().getAttempts().size()-1;` needs to be moved inside `if (t instanceof UserRecordFailedException)`

Reply:
Done